src/Ardalis.Specification/Specification.cs

```diff
@@ -84,10 +84,30 @@ public class Specification<T> : ISpecification<T>
 
     // Specs are not intended to be thread-safe, so we don't need to worry about thread-safety here.
     internal void Add(WhereExpressionInfo<T> whereExpression) => (_whereExpressions ??= new(2)).Add(whereExpression);
-    internal void Add(SearchExpressionInfo<T> searchExpression) => (_searchExpressions ??= new(2)).Add(searchExpression);
     internal void Add(OrderExpressionInfo<T> orderExpression) => (_orderExpressions ??= new(2)).Add(orderExpression);
     internal void Add(IncludeExpressionInfo includeExpression) => (_includeExpressions ??= new(2)).Add(includeExpression);
     internal void Add(string includeString) => (_includeStrings ??= new(1)).Add(includeString);
+    internal void Add(SearchExpressionInfo<T> searchExpression)
+    {
+        if (_searchExpressions is null)
+        {
+            _searchExpressions = new(2) { searchExpression };
```

Comment:
Replace magic number `2` with meaningful named constant?

Reply:
This represents the initial capacity. And it can be different for each collection, for example we may choose for Where to be 4, for Search 2, etc. We need to introduce 6-7 constants, and not sure whether that makes it better or worse.

Reply:
Updated to use constants.